python/cuspatial/cuspatial/core/binops/intersection.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import cudf
+from cudf.core.column import arange, build_list_column
+
+from cuspatial._lib.intersection import (
+    pairwise_linestring_intersection as c_pairwise_linestring_intersection,
+)
+from cuspatial.core._column.geocolumn import GeoColumn
+from cuspatial.core.geoseries import GeoSeries
+from cuspatial.utils.column_utils import contains_only_linestrings
+
+
+def pairwise_linestring_intersection(
+    linestrings1: GeoSeries, linestrings2: GeoSeries
+):
+    """
+    Compute the intersection of two GeoSeries of linestrings.
+
+    Note
+    ----
+    Note that the result should be interpreted as type list<union>, since
+    cuDF does not support union type, the result is returned as a sperate list
+    and a GeoSeries.
+
+    Parameters
+    ----------
+    linestrings1 : GeoSeries
+        A GeoSeries of linestrings.
+    linestrings2 : GeoSeries
+        A GeoSeries of linestrings.
+
+    Returns
+    -------
+    Tuple[cudf.Series, GeoSeries, DataFrame]
+        A tuple of three elements:
+        - An integral cudf serial to indicate the offsets of the geoseries.
+        - A Geoseries of the results of the intersection.
+        - A DataFrame of the ids of the linestrings and segments that
+          the intersection results came from.
+    """
+
+    if len(linestrings1) == 0:
+        return (
+            cudf.Series([0]),
+            GeoSeries([]),
+            cudf.DataFrame(
+                {
+                    "lhs_linestring_id": [],
+                    "lhs_segment_id": [],
+                    "rhs_linestring_id": [],
+                    "rhs_segment_id": [],
+                }
+            ),
+        )
+
+    if not contains_only_linestrings(
+        linestrings1
+    ) or not contains_only_linestrings(linestrings2):
```

Comment:
This is very weird formatting. Can it not be:

```suggestion
    if not contains_only_linestrings(linestrings1) or 
       not contains_only_linestrings(linestrings2):
```

Reply:
Probably just what the `black` autoformatter did with it.

Reply:
Yes, this is due to `black`. And the reason why black does this is because of [PEP-8](https://peps.python.org/pep-0008/#maximum-line-length):

> The preferred way of wrapping long lines is by using Python’s implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.

Reply:
Found a better way to express this.